frontend/packages/topology/src/behavior/withCreateConnector.tsx

```diff
@@ -162,20 +165,21 @@ const CreateConnectorWidget: React.FC<CreateConnectorWidgetProps> = observer((pr
       {prompt && (
         <ContextMenu
           reference={{ x: prompt.event.pageX, y: prompt.event.pageY }}
+          className={contextMenuClass}
           open
           onRequestClose={() => {
             setActive(false);
             onKeepAlive(false);
           }}
         >
           {prompt.choices.map((c) => (
-            <ContextMenuItem
-              key={c.label}
-              onClick={() => {
-                onCreate(prompt.element, prompt.target, prompt.event, c);
-              }}
-            >
-              {c.label}
+            <ContextMenuItem key={c.label}>
+              {
+                <KebabItem
+                  option={c}
+                  onClick={() => onCreate(prompt.element, prompt.target, prompt.event, c)}
+                />
+              }
             </ContextMenuItem>
```

Comment:
We cannot change this to use a `KebabItem` because this is generic code that shouldn't depend on console.
We'll need to find a way to support the console use case without this dependency.
Doesn't this break the storybook sample?
If it doesn't break it then I suppose we can keep it with a TODO to fix later.
But otherwise we need a way to supply a list of react components that go into the context menu.

Maybe we simply add the ability to supply a react element as a choice instead:

```
choices: ConnectorChoice[] | React.ReactNode;
```

```
         {
           React.isValidElement(prompt.choices) ? prompt.choices : 
            prompt.choices.map((c) => (
            <ContextMenuItem
              key={c.label}
              onClick={() => {
                onCreate(prompt.element, prompt.target, prompt.event, c);
              }}
            >
              {c.label}
            </ContextMenuItem>
          ))}
```

Then in your `onCreate` function instead of returning the list of actions, you actually build the menu items like we already do elsewhere. You'd be able to use the kebab actions handling of the callback / href without duplicating the code.

Reply:
@nemesis09  Check should be on item, `React.isValidElement(prompt.choices?.[0]`

Reply:
Done